python/vyos/ifconfig/interface.py

```diff
@@ -371,14 +419,21 @@ def _delete(self):
         # NOTE (Improvement):
         # after interface removal no other commands should be allowed
         # to be called and instead should raise an Exception:
-        cmd = 'ip link del dev {ifname}'.format(**self.config)
+        netns = get_interface_namespace(self.config['ifname'])
+        netns_exec = f'ip netns exec {netns} ' if netns else ''
+        cmd = netns_exec
+        cmd += 'ip link del dev {ifname}'.format(**self.config)
         return self._cmd(cmd)
 
     def _set_vrf_ct_zone(self, vrf):
         """
         Add/Remove rules in nftables to associate traffic in VRF to an
         individual conntack zone
         """
+        # Don't allow for netns yet
```

Comment:
Maybe add an exception here and extend `vyos.configverify.verify_vrf()` that vrf and netns can't be used at the same time.

Then the user is properly informed and the `verify_vrf` handler is already installed at (hopefully) all required places.